enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx

```diff
@@ -72,41 +73,75 @@ export const SdkIframeEmbedSetupProvider = ({
     ...(settings.questionId && { questionId: Number(settings.questionId) }),
   });
 
-  const updateSettings = useCallback(
-    (nextSettings: Partial<SdkIframeEmbedSettings>) =>
-      setSettings(
-        (prevSettings) =>
-          ({
-            ...prevSettings,
-            ...nextSettings,
-          }) as SdkIframeEmbedSettings,
-      ),
-    [setSettings],
+  const fallbackDashboardId =
+    recentDashboards[0]?.id ?? EMBED_FALLBACK_DASHBOARD_ID;
+
+  const handleEmbedSettingsRestored = useCallback(
+    (settings: SdkIframeEmbedSettings | null) => {
+      if (isEmbedSettingsLoaded) {
+        return;
+      }
+
+      if (settings) {
+        setSettings(settings);
+      } else {
+        // Apply the default settings if the user settings are empty.
+        const defaults = getDefaultSdkIframeEmbedSettings(
+          "dashboard",
+          fallbackDashboardId,
+        );
+
+        setSettings(
+          (prev) => ({ ...prev, ...defaults }) as SdkIframeEmbedSettings,
+        );
+      }
+
+      setEmbedSettingsLoaded(true);
+    },
+    [fallbackDashboardId, isEmbedSettingsLoaded],
   );
 
-  useEffect(() => {
-    if (!isEmbedSettingsLoaded && !isRecentsLoading) {
-      const defaultSettings = getDefaultSdkIframeEmbedSettings(
-        "dashboard",
-        recentDashboards[0]?.id ?? EMBED_FALLBACK_DASHBOARD_ID,
-      );
+  const { storeSetting } = usePersistJsonViaUserSetting({
+    settingKey: "sdk-iframe-embed-setup-settings",
+    omitKeys: ["instanceUrl"],
+    onRestore: handleEmbedSettingsRestored,
+    debounceMs: PERSIST_EMBED_SETTINGS_DEBOUNCE_MS,
 
-      updateSettings(defaultSettings);
-      setEmbedSettingsLoaded(true);
-    }
-  }, [
-    isRecentsLoading,
-    isEmbedSettingsLoaded,
-    recentDashboards,
-    updateSettings,
-  ]);
+    // Wait until recents are loaded before restoring the settings.
+    // This ensures the fallback id contains the most recent resource.
+    skipRestore: isRecentsLoading,
```

Comment:
Without this, we ran into a race condition where loading the settings is faster than loading recents, and when the settings turned out to be empty we won't get the most recent dashboard as a fallback.